### PR TITLE
Hasura migrations

### PR DESCRIPTION
**Pull request by ckru-kiel**
- merged person and participant, person is now called user
- semester renamed to program
- added some field that were missing from the course table
- added table for locations
- reworked mail tables

### CODE REVIEW

File: backend/metadata/databases/default/tables/public_User.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  using:
+    foreign_key_constraint_on:
+      column: UserId
+      table:
+        name: Admin
+        schema: public
+- name: Attendences
+  using:
+    foreign_key_constraint_on:
+      column: PersonId
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/public_User.yaml]: Should we rename this column to `UserId`?

File: backend/metadata/databases/default/tables/public_User.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  using:
+    foreign_key_constraint_on:
+      column: UserId
+      table:
+        name: Enrollment
+        schema: public
+- name: Instructors
+  using:
+    foreign_key_constraint_on:
+      column: PersonId
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/public_User.yaml]: Should we rename this column to `UserId`?

File: backend/metadata/databases/default/tables/public_User.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  using:
+    foreign_key_constraint_on:
+      column: Sender_id
+      table:
+        name: Mail
+        schema: public
+- name: Referents
+  using:
+    foreign_key_constraint_on:
+      column: PersonId
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/public_User.yaml]: Should we rename this column to `UserId`?

File: backend/metadata/databases/default/tables/tables.yaml [hunk 4]
```diff
@@ -11,18 +11,18 @@
 - "!include public_EnrollmentStatus.yaml"
 - "!include public_Event.yaml"
 - "!include public_Instructor.yaml"
+- "!include public_Location.yaml"
 - "!include public_Mail.yaml"
-- "!include public_MailLog.yaml"
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/tables.yaml:15]: Why does this file still exists? Can we delete it?

File: backend/metadata/databases/default/tables/tables.yaml [hunk 5]
```diff
@@ -11,18 +11,18 @@
...
+- "!include public_MailTemplate.yaml"
 - "!include public_MotivationGrade.yaml"
 - "!include public_OnlineHybridOffline.yaml"
-- "!include public_Participant.yaml"
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/tables.yaml:18]: Why does this file still exists? Can we delete it?

File: backend/metadata/databases/default/tables/tables.yaml [hunk 6]
```diff
@@ -11,18 +11,18 @@
...
-- "!include public_Person.yaml"
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/tables.yaml:19]: Why does this file still exists? Can we delete it?

File: backend/metadata/databases/default/tables/tables.yaml [hunk 7]
```diff
@@ -11,18 +11,18 @@
...
+- "!include public_Program.yaml"
 - "!include public_Project.yaml"
 - "!include public_ProjectEnrollment.yaml"
 - "!include public_ProjectType.yaml"
 - "!include public_Referent.yaml"
-- "!include public_Semester.yaml"
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/tables.yaml:24]: Why does this file still exists? Can we delete it?

**Review by fschindler — CHANGES_REQUESTED** [review]
Looks quite good! I've suggested some more renamings and found some possibly unused files.

Should we rename the Relation `Admin.Person` to `Admin.User`?
Should we rename the Relation `Employment.People` to `Employment.Users`?

File: backend/metadata/databases/default/tables/tables.yaml (hunk 4 above)
Comment (ckru-kiel) [reply backend/metadata/databases/default/tables/tables.yaml:15]: Not sure why hasura itself doesnt delete these when the tables are removed. deleted them manually now. hope it doesnt cause problerms

File: backend/metadata/databases/default/tables/public_Mail.yaml [hunk 8]
```diff
@@ -4,12 +4,15 @@ table:
...
+- definition:
+    enable_manual: true
+    insert:
+      columns: "*"
+  name: Mail_Test
+  retry_conf:
+    interval_sec: 10
+    num_retries: 0
+    timeout_sec: 60
+  webhook: http://172.17.0.1:3000
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/public_Mail.yaml]: Should this event trigger be part of this PR or is it an accidentally committed test?

File: backend/metadata/databases/default/tables/public_Attendence.yaml [hunk 9]
```diff
@@ -2,19 +2,9 @@ table:
...
-select_permissions:
-- permission:
-    columns:
-    - Attending
-    - Id
-    filter:
-      Person:
-        AuthId:
-          _eq: X-Hasura-User-Id
-  role: user
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/public_Attendence.yaml:20]: Why did you removed these permissions? We need to fetch all Attendance Records associated with the current user to display the Attendance section on the course page.

File: backend/metadata/databases/default/tables/public_User.yaml [hunk 10]
```diff
@@ -55,7 +62,6 @@ select_permissions:
   role: anonymous
 - permission:
     columns:
-    - AuthId
```
Comment (fschindler) [inline backend/metadata/databases/default/tables/public_User.yaml]: We need to access this property for filtering e.g. Attendance Records.

**Review by fschindler — CHANGES_REQUESTED** [review]
The changes look quite good. But you probably accidentally introduced some other changes. Please have a look at my new comments.

**Comment by okwast** [comment]
@ckru-cbb I think you missed the renaming of the `Person` in the instructor to also be `User`.

**Comment by okwast** [comment]
I yesterday wrote a comment about some places not being changed from Person to User.
However after applying the migrations and meta data a second time things were correct so I deleted the comment.
But as you probably got a notification in some way I should probably have edited it.

TL/DR: Didn't find another issue, let's get this merged 👏

File: backend/metadata/databases/default/tables/public_Mail.yaml (hunk 8 above)
Comment (fschindler) [reply backend/metadata/databases/default/tables/public_Mail.yaml]: Removed this for the moment. If we need this, let's create a separate PR.

File: backend/metadata/databases/default/tables/public_Attendence.yaml (hunk 9 above)
Comment (fschindler) [reply backend/metadata/databases/default/tables/public_Attendence.yaml:20]: Reintroduced this.

File: backend/metadata/databases/default/tables/public_User.yaml (hunk 10 above)
Comment (fschindler) [reply backend/metadata/databases/default/tables/public_User.yaml]: Added this again.

**Review by fschindler — APPROVED** [review]

**Pull request merged**